### ISSUE TITLE
[apex] EmptyCatchBlock false positive with formal (doc) comments

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3953]
**Affects PMD Version:** 6.45.0

**Rule:** [EmptyCatchBlock](https://pmd.github.io/latest/pmd_rules_apex_errorprone.html#emptycatchblock)

**Description:** The current implementation ignores formal comments when the option `allowCommentedBlocks` is used. Only single line comments or multi line comments, which are not doc comments, are considered. Therefore, if there is only a doc comment in the catch block, it is considered to be empty and is reported - despite the fact, that here is a comment and the block is not empty.

This might be a false positive.

However, current test cases explicitly verify this case as being reported.

Overall, this might be a contrived example...

Background: the behaviour of this rule might be changed in pmd7 with #3893

**Code Sample demonstrating the issue:**

```apex
@IsTest
private class FunctionalityTest {
    static testMethod void testWithModifier() {
        try {
        } catch (Exception e) { // line 5
          /** NOK: doc comment inside of empty catch block; should be reported */
        }
}
```

Complete test cases:
https://github.com/pmd/pmd/blob/eae87d26a7bce1ae09e841130de320f635d48921/pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/errorprone/xml/EmptyCatchBlock.xml#L294-L319

**Expected outcome:**

PMD should not report a violation at line 5. This is a false-positive.

**Comment by adangel** [comment]
@rsoesemann @SCWells72 What do you think? Should this be a violation or not?

**Comment by oowekyala** [comment]
For context, the reason I changed this in #3893 is that the rule uses the attribute `@ContainsComment`, and I think if this attribute doesn't consider doc comments comments, then it is lying (=bugged). The change is also a code simplification. If we want to preserve the current behaviour in PMD 7, then we'll need to go out of our way to do so.
